### PR TITLE
PPC 2.0: Thumbnail Edge Case Poster Image Backup

### LINKED ISSUE #8070: PPC 2.0: Thumbnail Edge Case Poster Image Backup
Screen Shot 2021-06-23 at 12 48 19 PM

We have 1 check in the PPC for videos that have no poster image. We're supposed to use the poster image as the thumbnail in the card for the issue! Clearly that can't happen with this situation. 

We need to use a placeholder image as the `displayBackground`  when there's no `displayBackground` passed through a card to a thumbnail. 

This ticket is for updating the thumbnail to do this and the `videoElementMissingPoster` (which is contingent on this PR #8042 ) 

TBD: placeholder image source

### PR DESCRIPTION
**Pull request by samwhale**
## Context

PPC 2.0 + [slack convo about backups if no poster image](https://xwp.slack.com/archives/C01ESBB8ZN1/p1624921376196400?thread_ts=1624477840.171600&cid=C01ESBB8ZN1)

## Summary

Show the first frame of the video as an image if there is no `poster` on the resource in the PPC

## Relevant Technical Choices

Changed this at the `LayerIcon` level, so this change will also go into effect in the layer panel since the `LayerIcon` is a shared component.

~If we only want this to be applied to the cards in the PPC just let me know and I can update that.~ Updated it so the ppc is the only thing that is changed.

## To-do

n/a

## User-facing changes

||Before|After|
|--|--|--|
|Checklist|image|image|

## Testing Instructions

Easiest way would be to set `iconImage` in `assets/src/edit-story/elements/video/icon.js` to `false`. Should see it in action that way

Otherwise, the best way would be to upload a video and remove the poster image from the video resource.

### QA

- [x] This is a non-user-facing change and requires no QA

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8070

### CODE REVIEW

**Comment by samwhale** [comment]
Looks like percy caught another change:

|Before|After|
|--|--|
|image|image|

This one doesn't look as good. I'm thinking of making it so that the cards are the only change instead of the whole app. Thoughts @BrittanyIRL @embarks?

**Comment by samwhale** [comment]
> This one doesn't look as good. I'm thinking of making it so that the cards are the only change instead of the whole app. Thoughts @BrittanyIRL @embarks?

I ended up making it so that the card is the only one that shows the video preview:

File: assets/src/edit-story/components/checklist/checks/imageElementResolution.js [hunk 1]
```diff
@@ -107,7 +107,9 @@ const ImageElementResolution = () => {
                 key={element.id}
                 onClick={() => handleClick(element.id)}
                 type={THUMBNAIL_TYPES.IMAGE}
-                displayBackground={<LayerThumbnail page={element} />}
+                displayBackground={
+                  <LayerThumbnail page={element} showVideoPreviewAsBackup />
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/checklist/checks/imageElementResolution.js]: If this is an image check, do we need to worry about the video backup preview? Same question for the checks re text.

File: assets/src/edit-story/components/thumbnail/components/LayerThumbnail.js [hunk 2]
```diff
@@ -22,11 +22,17 @@ import PropTypes from 'prop-types';
...
 import { getDefinitionForType } from '../../../elements';
 
-export const LayerThumbnail = ({ page }) => {
+export const LayerThumbnail = ({ page, showVideoPreviewAsBackup }) => {
   const { LayerIcon } = getDefinitionForType(page?.type);
-  return <LayerIcon element={page} />;
+  return (
+    <LayerIcon
+      element={page}
+      showVideoPreviewAsBackup={showVideoPreviewAsBackup}
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/thumbnail/components/LayerThumbnail.js]: If you're passing this into the majority of the checks using `showVideoPreviewAsBackup` should it just become the default here to clear up props.

File: assets/src/edit-story/elements/media/videoImage.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+`;
+
+function VideoImage({ alt, ...attrs }) {
+  return (
+    <Video
+      controls={false}
+      crossOrigin="anonymous"
+      disablePictureInPicture
+      muted
+      noControls
```
Comment (BrittanyIRL) [inline assets/src/edit-story/elements/media/videoImage.js:38]: Nice!

**Review by BrittanyIRL — COMMENTED** [review]
This works great! The video shows as an image in the checklist and the existing layerIcon is left as is for the layers panel. 

If @miina thinks this is useful, we could make a new ticket to extend this same fallback to the layers panel for consistency. 

Just that one question about making `showVideoPreviewAsBackup` default in the `LayerThumbnail` if it's getting applied to every element type / why it's needed on text and image and page type checks. 
Screen Shot 2021-06-30 at 9 58 38 AM

File: assets/src/edit-story/components/thumbnail/components/LayerThumbnail.js (hunk 2 above)
Comment (embarks) [reply assets/src/edit-story/components/thumbnail/components/LayerThumbnail.js]: I think if there are `video` element types, we could set this to true for the `LayerThumbnail`

Comment (samwhale) [reply assets/src/edit-story/components/thumbnail/components/LayerThumbnail.js]: Great catch! Just updated that

**Comment by miina** [comment]
> If @miina thinks this is useful, we could make a new ticket to extend this same fallback to the layers panel for consistency.

Thanks for tagging me, would be nice for consistency, in this case it would also influence the Filters panel and the Page background panel for background video.

Also saw that Omar commented this in Slack:
> I think here a placeholder image may make sense, something not too pretty/cute so it really pushes the point that setting a poster image is beneficial 

I assume this goes for the PPC only? I'll follow up on this.

**Comment by samwhale** [comment]
> > If @miina thinks this is useful, we could make a new ticket to extend this same fallback to the layers panel for consistency.
> 
> Thanks for tagging me, would be nice for consistency, in this case it would also influence the Filters panel and the Page background panel for background video.
> 
> Also saw that Omar commented this in Slack:
> 
> > I think here a placeholder image may make sense, something not too pretty/cute so it really pushes the point that setting a poster image is beneficial
> 
> I assume this goes for the PPC only? I'll follow up on this.

Yes, this is only for the PPC. I made it so that it wouldn't affect the layer's panel and the filter's panel, but it would be a quick update if we wanted that later.

File: assets/src/edit-story/components/thumbnail/components/LayerThumbnail.js [hunk 4]
```diff
@@ -24,7 +24,14 @@ import { getDefinitionForType } from '../../../elements';
 
 export const LayerThumbnail = ({ page }) => {
   const { LayerIcon } = getDefinitionForType(page?.type);
-  return <LayerIcon element={page} />;
+
+  const props = { element: page };
+
+  if (page?.type === 'video') {
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/thumbnail/components/LayerThumbnail.js:30]: smart!

**Review by BrittanyIRL — APPROVED** [review]
Thanks for the updates!

**Review by embarks — APPROVED** [review]

**Pull request merged**